File: Custodial.BoilerPlate.Core/Custodial.BoilerPlate.Core/Service Settings/ServiceSettings.cs
using Custodial.BoilerPlate.Core.Interfaces;
using Newtonsoft.Json;
using System.Reflection;
using System.Threading.Tasks;

namespace Custodial.BoilerPlate.Core.Service_Settings
{
    public class ServiceSettings : IServiceSettings
    {
        [JsonIgnore]
        private IResourceLoader resourceLoader = new ResourceLoader();
        [JsonConverter(typeof(ConcreteTypeConverter<NetworkSettings>))]
        public INetworkSettings networkSettings { get; set; } = new NetworkSettings();

        [JsonConverter(typeof(ConcreteTypeConverter<DatabaseSettings>))]
        public IDatabaseSettings databaseSettings { get; set; } = new DatabaseSettings();

        [JsonConverter(typeof(ConcreteTypeConverter<CustodialAddressingSettings>))]
        public ICustodialAddressingSettings casSettings { get; set; } = new CustodialAddressingSettings();

        [JsonIgnore]
        public string certSSL = $"cert";

        [JsonIgnore]
        public string passwordSSL = $"pass";

        public ServiceSettings()
        {

        }

        public Task InitServiceSettingsAsync( string resourceFile, Assembly assembly)
        {
            if (assembly == null)
            {
                assembly = this.GetType().Assembly;
            }
            string embeddedString = resourceLoader.GetEmbeddedResourceString(assembly, resourceFile);
            var temp = JsonConvert.DeserializeObject<ServiceSettings>(embeddedString);
            networkSettings = temp.networkSettings;
            databaseSettings = temp.databaseSettings;
            casSettings = temp.casSettings;
            networkSettings.InitNetworkSettingsAsync();
            return Task.CompletedTask;
        }
    }
}

File: FrontEndv3/Webserverv2/src/Classes/CustodialAddressingServices/ServiceSettings.ts
import {IServiceSettings} from '../../Interfaces/CustodialAddressingServices/IServiceSettings';
import {INetworkSettings} from '../../Interfaces/CustodialAddressingServices/INetworkSettings';
import {IDatabaseSettings} from '../../Interfaces/CustodialAddressingServices/IDatabaseSettings';
import {ICasSettings} from '../../Interfaces/CustodialAddressingServices/ICasSettings';

export class ServiceSettings implements IServiceSettings{
    public networkSettings: INetworkSettings; 
    public databaseSettings: IDatabaseSettings;
    public casSettings: ICasSettings;

    constructor(networkSettings: INetworkSettings, databaseSettings: IDatabaseSettings,
         casSettings: ICasSettings){
        this.networkSettings = networkSettings;
        this.casSettings = casSettings;
        this.databaseSettings = databaseSettings;
    }


}